docs/architecture.md

```diff
@@ -4,213 +4,217 @@
+  - we can have bigger intervals between each run compared to a maximum timeout of **300 seconds** for a queue consumer, enabling us to decrease number of requests
+  - we can keep track not only of deal resolution, but also instrument alerts when a status is `OFFERED` for more time than expected
+
+## Data Stack
+
+The Data stack is responsible for the state of the w3filecoin.
+
+It keeps track of the received pieces, submitted aggregates, as well as in which aggregate a given piece is (inclusion). In addition, it must keep the necessary state for the `buffer-queue` to operate, as well as the state of a given aggregate over time until a deal is fulfilled. It is worth mentioning, that keeping track of problematic pieces that could not be added to an aggregate should also be properly tracked.
+
+To achieve required state management, we will be relying on a S3 Bucket and a dynamoDB table as follows:
+
+| store           | type     | key                      | expiration |
+|-----------------|----------|--------------------------|------------|
+| piece-store     | DynamoDB | `piece`                  | ---        |
+| buffer-store    | S3       | `${blockCid}/{blockCid}` | 30 days    |
+| aggregate-store | DynamoDB | `piece`                  | ---        |
+| inclusion-store | DynamoDB | `aggregate`              | ---        |
+
+### Queries and associayed stores
+
+1. Check if piece is already in the pipeline (`inclusion-store` with fallback for `piece-store`)
+  - Before getting a piece into the pipelined queue, it should be stored to guarantee uniqueness
+  - Also important to be able to report back on the "in progress" work when deal state of a piece is still unknown by fallbacking to Head Request in bucket
+2. Put and Get buffer blocks (`buffer-store`)
+  - While `buffer-queue` is concatenating buffers, these blocks will be stored to be propagated through queue stages via their CIDs
+  - Key `${blockCid}/{blockCid}`, Value empty with expiration date
+3. Put aggregates offered to `spade` and get aggregates pending resolution (`aggregate-store`)
+4. Write pieces together with the aggregate they are part of (`inclusion-store`)
+5. Get deal state of a given piece (`inclusion-store` with fallback for `piece-store`)
+  - Rely on DynamoDB to get `aggregate` where the piece is (secondary index) and use it ask Spade for details for the deal
+  - In case there is no record in Dynamo, we can fallback to check S3 bucket, in order to reply that piece is being aggregated if it is there
+
+TODO: metrics
+
+### `piece-store` schema
+
+```typescript
+interface piece {
+  // CID of the piece `bagy...content` (primary index, partition key)
+  piece: PieceCID
+  // number of milliseconds elapsed since the epoch for piece inserted
+  insertedAt: number
+  // identifier of producer to derive `group-id` `did:web:web3.storage` (primary index, sort key)
+  space: string
+  // identifier of group to derive `group-id` `did:web:free.web3.storage`
+  group: string
```

Comment:
I'm confusing as to what is `space` and what is `group` and how are they different. Also perhaps we should use something other than `space` as a field name to avoid confusion with user spaces.

Reply:
`space` is the tenant, we could call it tenant like spade does. Group are the different SLA plans that the tenant has